File: app/src/main/java/com/example/shenjack/zhihudailyreader/storydetail/DetailContract.java
package com.example.shenjack.zhihudailyreader.storydetail;

import com.example.shenjack.zhihudailyreader.BaseView;
import com.example.shenjack.zhihudailyreader.data.Detail;

/**
 * Created by ShenJack on 2017/6/17.
 */

public interface DetailContract {
    interface View extends BaseView<Presenter>{
        void showLoadingIndicator(boolean visible);

        void showStoryDetail(Detail detail);

    }

    interface Presenter {
        void loadStoryDetail(String id);

    }
}

File: app/src/main/java/com/example/shenjack/zhihudailyreader/storydetail/DetailContract.kt
package com.example.shenjack.zhihudailyreader.storydetail

import com.example.shenjack.zhihudailyreader.BaseView
import com.example.shenjack.zhihudailyreader.data.Detail

/**
 * Created by ShenJack on 2017/6/17.
 */

interface DetailContract {
    interface View : BaseView<Presenter> {
        fun showLoadingIndicator(visible: Boolean)

        fun showStoryDetail(detail: Detail)

    }

    interface Presenter {
        fun loadStoryDetail(id: String)

    }
}
